cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/util/AwsIamService.java

```diff
@@ -235,14 +235,14 @@ public Policy getPolicy(String policyFileName, Map<String, String> replacements)
      * @return list of evaluation results
      */
     public List<EvaluationResult> validateRolePolicies(AmazonIdentityManagement iam, Role role,
-            Collection<Policy> policies) {
+            Collection<Policy> policies) throws AmazonIdentityManagementException {
         List<EvaluationResult> evaluationResults = new ArrayList<>();
         for (Policy policy : policies) {
             for (Statement statement : policy.getStatements()) {
```

Comment:
I merged it, but there are 4 embedded "for cycles" still:
```
        for (Policy policy : policies) {	        for (Policy policy : policies) {
            for (Statement statement : policy.getStatements()) {
...
                         for (Policy policy : policies) {
            for (Statement statement : policy.getStatements()) {
                SortedSet<String> actions = getStatementActions(statement);
                SortedSet<String> resources = getStatementResources(statement);
                evaluationResults.addAll(simulatePrincipalPolicy(iam, role.getArn(),
                        actions, resources));
            }
        }
```
Can we optimize this like we did here?

https://github.com/hortonworks/cloudbreak/blob/e98a3317bfba3731ba23605f4ecef1869acf89fa/cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCredentialVerifier.java#L75
https://github.com/hortonworks/cloudbreak/blob/e98a3317bfba3731ba23605f4ecef1869acf89fa/cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCredentialVerifier.java#L60